src/vehicle.cpp

```diff
@@ -177,35 +177,15 @@ void VehicleServiceInDepot(Vehicle *v)
 }
 
 /**
- * Check if the vehicle needs to go to a depot in near future (if a opportunity presents itself) for service or replacement.
- *
- * @see NeedsAutomaticServicing()
- * @return true if the vehicle should go to a depot if a opportunity presents itself.
+ * Test whether there is some pending autoreplace.
+ * @note We do this after the service-interval test.
+ *       There are a lot more reasons for autoreplace to fail than we can test here reasonably.
+ * @see NeedsServicing()
+ * @return true if there is a pending autoreplace
  */
-bool Vehicle::NeedsServicing() const
+bool Vehicle::HasPendingReplace() const
```

Comment:
This is not a critique, but I think the diff will look nicer if you move `HasPendingReplace` to below ´NeedsServicing` :)